article/sections/_5_graph_testing.tex

```diff
@@ -1,14 +1,230 @@
+In particular, we created simulations to assess our mean-independence testing procedure.
+We assessed the performance of our mean-independence testing procedures using simulations where $Y \leftarrow Z \rightarrow X$ and $X$ either did or did not cause $Y$.
+
+Of particular importance were our simulations under the null hypothesis where $X$ was conditionally independent of $Y$.
+Our initial simulations used random forests as our conditional expectation models and permutations as resampling methods.
+Random forests are a non-parametric method that would allow us to have less fear of model misspecification, and permutations are easy to implement.
+However, under the null hypothesis, we discovered that the tests based on the random forest models did not result in p-values that were uniformly distributed.
+Given that we planned to use these procedures in a manner akin to hypothesis-testing, we hoped that our test statistics would be U-statistics.
+When we switched from the combination of random forests and permutations to linear regressions and permutations, our p-values were indeed empirically, uniformly distributed.
+Moreover, we still retained high power.
+
+We do not claim that these choices for assessing mean independence will always be appropriate.
+Indeed, one should assess one's tests on simulated data that resembles one's real data.
+For our illustrative purposes though, the combination of linear regressions, permutations, and $R^2$ resulted in adequate tests of marginal and conditional mean-independence.
+
+\subsubsection{Testing assumptions involving latent variables}
+\label{sec:latent-tests}
+% Describe latent variable testing
+Now that we have described testing with observable variables, we can more easily describe conditional independence tests that involve unobserved (i.e., latent) variables.
+Indeed, when dealing with observational data, we will frequently find ourselves not having observed all variables that are of interest.
+Nevertheless, we still wish to test whether our data contradicts our graph.
+One way to directly extend our conditional independence testing to account for latent variables is to adopt a missing data perspective and impute the latent variables from a prior distribution.
+In particular, we can generalize our previous tests as follows.
+
+First, we can consider that instead of performing one test with a set of observed $X$, observed $Y$ and observed $Z$, we instead perform tests of observed $X$, observed $Y$, and imputed $Z$.
+This recasts the randomness underlying the null-distribution in our original test statistic of $R^2$ as a function of our permutation of $Y$ and our imputation of $Z$.
+Here, we impute $Z$ by sampling from the prior or posterior distribution of $Z$, depending on whether we're testing independencies before or after performing inference on our model's parameters.
```

Comment:
I feel this subsection needs more information on the process of imputing Z. Even to me, it's still unclear how one would have a prior or posterior on Z if it's unobserved. I can understand having a prior distribution (if one theorizes on what Z is), but even that would be a marginal distribution of Z as opposed to a conditional one wrt X or Y (or a joint distribution of Z, X, Y). I can't really understand what a posterior distribution of Z means -- posterior based on what model of Z?